lib/pyfrc/physics/drivetrains.py

```diff
@@ -104,41 +104,41 @@ def mecanum_drivetrain(lr_motor, rr_motor, lf_motor, rf_motor, wheel_size=(8/12)
         :param rr_motor:   Right rear motor value (-1 to 1); 1 is forward
         :param lf_motor:   Left front motor value (-1 to 1); 1 is forward
         :param rf_motor:   Right front motor value (-1 to 1); 1 is forward
-        :param wheel_size: Size of robot wheels
+        :param x_wheelbase: The distance in feet between right and left wheels.
+        :param y_wheelbase: The distance in feet between forward and rear wheels.
         :param speed:      Speed of robot in feet per second (see above)
         
         :returns: Speed of robot in x (ft/s), Speed of robot in y (ft/s), 
                   clockwise rotation of robot (radians/s)
     '''
-    
-    lr = lr_motor * speed
-    rr = rr_motor * speed
-    lf = lf_motor * speed
-    rf = rf_motor * speed
-    
+
     #
     # From http://www.chiefdelphi.com/media/papers/download/2722 pp7-9
     # [F] [omega](r) = [V]
     #
-    # F is 
+    # F is
     # .25  .25  .25 .25
     # -.25 .25 -.25 .25
     # -.25k -.25k .25k .25k
     #
     # omega is
     # [lf lr rr rf]
+
+    # Calculate speed of each wheel
+    lr = lr_motor * speed
+    rr = rr_motor * speed
+    lf = lf_motor * speed
+    rf = rf_motor * speed
+
+    # Calculate K
+    k = abs(x_wheelbase/2) + abs(y_wheelbase/2)
+
+    # Calculate resulting motion
+    Vy = .25 * (lf + lr + rr + rf)
+    Vx = .25 * (lf + -lr + rr + -rf)
+    Vw = (.25/k) * (lf + lr + -rr + -rf)
     
-    r = wheel_size
-    
-    Vy = r * .25 * (lf + lr + rr + rf)
-    Vx = -r * .25 * (-lf + lr + -rr + rf)
-    
-    # For best results, k must be > 1
-    k = max(abs(Vx) + abs(Vy), 1.0)
-    
-    Vw = r * (.25/k) * (-lf + -lr + rr + rf)
-    
-    return Vx, Vy, -Vw
+    return Vx, Vy, Vw
```

Comment:
No, I inverted the signs on line 139 so that the additional negation was unnecessary.